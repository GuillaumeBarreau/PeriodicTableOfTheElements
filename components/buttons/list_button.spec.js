import { shallowMount } from '@vue/test-utils'
import cmp from '~/components/buttons/list_buttons.vue'

jest.mock('~/static/icons-menu.svg', () => 'ImageMoched')
describe('testing component List_buttons', () => {
  const wrapper = shallowMount(cmp, {
    propsData: {
      tableElements: [
        { groupBlock: 'name test_A' },
        { groupBlock: 'name test_B' }
      ]
    }
  })

  it('snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component Validation', () => {
    it('Should return the component list_button.vue', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })
  })

  describe('Content validation', () => {
    it('should render the correct text', () => {
      expect(wrapper.text()).toContain('name test_A')
    })

    it('should render the correct text', () => {
      expect(wrapper.text()).toContain('name test_B')
    })
  })
})
