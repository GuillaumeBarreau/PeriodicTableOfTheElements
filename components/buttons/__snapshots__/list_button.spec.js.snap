// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component List_buttons snapshot testing 1`] = `
"<ul class=\\"content--button\\"><li class=\\"item--button\\"><div class=\\"item--button--icon --name_test_A\\">
      IC
    </div> <button-element-stub class=\\"item--button--label item--button--name_test_A\\">
      name test_A
    </button-element-stub></li><li class=\\"item--button\\"><div class=\\"item--button--icon --name_test_B\\">
      IC
    </div> <button-element-stub class=\\"item--button--label item--button--name_test_B\\">
      name test_B
    </button-element-stub></li></ul>"
`;
